src/c_kzg_4844.c

```diff
@@ -1755,37 +1641,46 @@ C_KZG_RET load_trusted_setup(
     const uint8_t *g2_bytes,
     size_t n2
 ) {
-    uint64_t i;
-    blst_p2_affine g2_affine;
-    g1_t *g1_projective = NULL;
     C_KZG_RET ret;
 
-    out->fs = NULL;
     out->g1_values = NULL;
     out->g2_values = NULL;
 
     /* Sanity check in case this is called directly */
     CHECK(n1 == TRUSTED_SETUP_NUM_G1_POINTS);
     CHECK(n2 == TRUSTED_SETUP_NUM_G2_POINTS);
 
+    /* It's the smallest power of 2 >= n1 */
+    unsigned int max_scale = 0;
+    while ((1ULL << max_scale) < n1)
+        max_scale++;
+
+    out->max_width = 1ULL << max_scale;
```

Comment:
I probably should assign to `out->max_width` at the end of the function instead to avoid mutating the output when the function fails?

Reply:
No strong opinion. As long as we document it, I think it's fine to mutate even when the function fails.

Reply:
This will be set to zero if it fails now. Do you still want to document this?

Reply:
I think it's OK either way.